distutils/tests/test_install.py

```diff
@@ -122,6 +124,18 @@ def cleanup():
         self.assertIn('userbase', cmd.config_vars)
         self.assertIn('usersite', cmd.config_vars)
 
+        actual_headers = os.path.relpath(cmd.install_headers, self.user_base)
+        if os.name == 'nt':
+            expect_headers = os.path.join(
+                os.path.relpath(os.path.dirname(self.old_user_site), self.old_user_base),
+                'Include',
+                'xx',
+            )
+        else:
+            expect_headers = os.path.join(sysconfig.get_python_inc(0, ''), 'xx')
```

Comment:
I'm struggling to understand the intention behind divergent tests here. It's annoying that the tests need to have such intimate knowledge of the implementation details in order to set expectations. Is there a better way that derives this expectation from the API?

Is it possible the expected value could be derived from some value in `sysconfig` (not `distutils.sysconfig`)? I'm trying to move more of the logic away from `distutils.sysconfig` as that module is deprecated, so this line moves in the wrong direction.

Reply:
It's because Windows doesn't include the runtime version in its headers subdirectory (`{prefix}\Include\xx`), while POSIX does (`{prefix}\include\python3.10\xx`).

But POSIX also seems to include the ABI flags in the name? Sometimes? I'm sure the logic is "fine", but I couldn't figure it out. Pretty sure it's also changed upstream with rebasing onto `sysconfig`, though not in any way that impacts default builds after 3.7.

If you weren't running 3.7 tests on this repo, I wouldn't even have noticed the change and would've stuck with my simpler test logic :)